pydantic/types.py

```diff
@@ -1055,3 +1056,74 @@ def validate(cls, value: date) -> date:
                 raise errors.DateNotInTheFutureError()
 
             return value
+
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ NANO TIME TYPE ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+TIME_UNITS = {
+    'ns': 1,
+    'ms': 10 ** 6,
+    's': 10 ** 9,
+    'm': 60 * 10 ** 9,
+    'h': 60 * 60 * 10 ** 9,
+    'd': 24 * 60 * 60 * 10 ** 9
+}
+time_string_re = re.compile(r'\s*(\d+\.?\d*)\s*([a-zA-Z]+)?', re.IGNORECASE)
```

Comment:
```suggestion
time_string_re = re.compile(r'\s*(\d+(?:\.\d+)?)\s*([a-z]{1,2})?', re.IGNORECASE)
```

Unless you think `1. ns` should be allowed?

Reply:
Also can you use `re.match` in the function instead of `re.compile` here, see #1132